sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,297 @@
+{
+	if (reader == NULL) {
+		return;
+	}
+
+	fclose(reader->scenario_file);
+	free(reader->lineBuf);
+}
+
+
+static long long reader_getFieldLong(const char *actField, char **nextField, int *error)
+{
+	char *endptr;
+	long long result = strtoll(actField, &endptr, 10);
+
+	if (!(*endptr == READER_DATA_SEPARATOR || *endptr == '\0' || *endptr == '\n')) {
+		*error = -1;
+		return -1;
+	}
+
+	if (*endptr == READER_DATA_SEPARATOR) {
+		*nextField = endptr + 1;
+	}
+
+	return result;
+}
+
+
+static int reader_accelDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	int err = 0;
+
+	actField = startOfDataSection;
+
+
+	result->type = SENSOR_TYPE_ACCEL;
+	result->timestamp = timestamp;
+
+	result->accels.accelX = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	result->accels.accelY = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	result->accels.accelZ = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	return 0;
+}
+
+
+static int reader_baroDataParse(char *startOfDataSection, time_t timestamp, sensor_event_t *result)
+{
+	char *actField, *nextField;
+	int err = 0;
+
+	actField = startOfDataSection;
+
+
+	result->type = SENSOR_TYPE_BARO;
+	result->timestamp = timestamp;
+
+	result->baro.pressure = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	result->baro.temp = reader_getFieldLong(actField, &nextField, &err);
+	if (err != 0) {
+		return -1;
+	}
+	actField = nextField;
+
+	return 0;
+}
+
+
+static time_t reader_nextTimeStampGet(simsens_time_t *timeStruct, time_t timestampFromFile)
+{
+	if (timeStruct->firstTimestamp == READER_TIMESTAMP_NOT_SET) {
+		timeStruct->firstTimestamp = timestampFromFile;
+		timeStruct->lastTimestamp = timestampFromFile;
+	}
+	else if (timeStruct->firstTimestamp == timestampFromFile) {
+		timeStruct->loops++;
+	}
+	else if (timeStruct->lastTimestamp < timestampFromFile) {
+		timeStruct->lastTimestamp = timestampFromFile;
+	}
+
+	return timestampFromFile + timeStruct->loops * (timeStruct->lastTimestamp - timeStruct->firstTimestamp);
+}
+
+
+int reader_read(simsens_reader_t *rd, event_queue_t *queue)
+{
+	char *actField, *nextField;
+	int sensorID, err = 0;
+	size_t curLineLen;
+	time_t timestamp, first_timestamp = READER_TIMESTAMP_NOT_SET;
+	sensor_event_t parsed_result;
+
+	while (eventQueue_full(queue)) {
+		curLineLen = getline(&rd->lineBuf, &rd->bufLen, rd->scenario_file);
+		if (curLineLen == -1) {
+			if (!feof(rd->scenario_file)) {
+				err = -1;
+				break;
+			}
+
+			/* Going to the beginning of the file */
+			if (fseek(rd->scenario_file, 0, SEEK_SET) != 0) {
+				err = -1;
+				break;
+			}
+
+			if (reader_skipHeader(rd) != 0) {
+				err = -1;
+				break;
+			}
+
+			continue;
```

Comment:
This `continue` has potential of endlessly spin the loop if there is some error in read and no data is pushed to queue. Some cap at failed reads is necessary.

Note: MISRA demands loops not bounded to be bounded with some cap